File: src/types/note.ts
export type NoteCategory =
  | 'All'
  | 'Ideas'
  | 'Plot'
  | 'Dialogue'
  | 'Research'
  | 'Scene'
  | 'Character'
  | 'Worldbuilding'
  | 'TODO'
  | 'Random'
  | 'Other'
  | 'Archived';

export interface Note {
  id: string;
  project_id: string;
  category: string;
  title: string;
  content: string;
  tags: string | null;
  archived_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateNoteInput {
  project_id: string;
  category?: string | null;
  title: string;
  content?: string | null;
  tags?: string | null;
}

export interface UpdateNoteInput {
  category?: string | null;
  title?: string | null;
  content?: string | null;
  tags?: string | null;
  archived?: boolean | null;
}

File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Note {
    pub id: String,
    pub project_id: String,
    pub category: String,
    pub title: String,
    pub content: String,
    pub tags: Option<String>,
    pub archived_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateNoteInput {
    pub project_id: String,
    pub category: Option<String>,
    pub title: String,
    pub content: Option<String>,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateNoteInput {
    pub category: Option<String>,
    pub title: Option<String>,
    pub content: Option<String>,
    pub tags: Option<String>,
    pub archived: Option<bool>,
}
